core/codenvy-hosted-api-workspace/src/main/java/com/codenvy/api/workspace/server/filters/RecipeScriptDownloadPermissionFilter.java

```diff
@@ -0,0 +1,55 @@
+/*
+ *  [2012] - [2016] Codenvy, S.A.
+ *  All Rights Reserved.
+ *
+ * NOTICE:  All information contained herein is, and remains
+ * the property of Codenvy S.A. and its suppliers,
+ * if any.  The intellectual and technical concepts contained
+ * herein are proprietary to Codenvy S.A.
+ * and its suppliers and may be covered by U.S. and Foreign Patents,
+ * patents in process, and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from Codenvy S.A..
+ */
+package com.codenvy.api.workspace.server.filters;
+
+import org.eclipse.che.api.core.ApiException;
+import org.eclipse.che.api.core.ForbiddenException;
+import org.eclipse.che.api.machine.server.MachineService;
+import org.eclipse.che.commons.env.EnvironmentContext;
+import org.eclipse.che.commons.subject.Subject;
+import org.eclipse.che.everrest.CheMethodInvokerFilter;
+import org.everrest.core.resource.GenericMethodResource;
+
+import static com.codenvy.api.workspace.server.WorkspaceDomain.DOMAIN_ID;
+import static com.codenvy.api.workspace.server.WorkspaceDomain.USE;
+
+/**
+ * Restricts access to methods of {@link org.eclipse.che.ide.ext.machine.server.RecipeScriptDownloadService} by users' permissions
+ *
+ * @author Mihail Kuznyetsov.
+ */
+public class RecipeScriptDownloadPermissionFilter extends CheMethodInvokerFilter {
```

Comment:
Are you sure that it works OK? I think NO. Because you missed @Filter and @Path annotations. Please add them and add tests for ensuring that this filter covers RecipeScriptDownloadService methods. You can take a look https://github.com/codenvy/codenvy/blob/master/core/codenvy-hosted-api-workspace/src/test/java/com/codenvy/api/workspace/server/filters/WorkspacePermissionsFilterTest.java as example